docs/notebooks/JAX_for_the_impatient.ipynb

```diff
@@ -0,0 +1,1293 @@
+{
```

Comment:
<p>Maybe add, "if you deal with DeviceArrays then computation is done on device so no additional data transfer is required, e.g. <code>jnp.dot(long_vector, long_vector)</code>  will only transfer a single scalar back from device to host.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/flax/pull/542/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Reply:
<p>Good point indeed, will add!</p>